tests/integration_tests/without_db/udfs/test_question_answering_udf.py

```diff
@@ -94,27 +96,15 @@ def test_question_answering_udf(
     result_df = ctx.get_emitted()[0][0]
     new_columns = ['answer', 'score', 'rank', 'error_message']
 
-    # assertions
-    is_score_typed_as_float = result_df['score'].dtypes == 'float'
-    is_rank_typed_as_int = result_df['rank'].dtypes == 'int'
-    is_error_message_none = not any(result_df['error_message'])
-    has_valid_shape =  \
-        result_df.shape == (n_rows * top_k, len(columns)+len(new_columns)-1)
-    has_valid_column_number = \
-        result_df.shape[1] == len(columns) + len(new_columns) - 1
-    is_rank_correct = \
-        all([result_df[row*top_k: top_k + row*top_k]
-            .sort_values(by='score', ascending=False)['rank']
-            .is_monotonic for row in range(n_rows)])
-
-    assert all((
-        is_score_typed_as_float,
-        is_rank_typed_as_int,
-        is_error_message_none,
-        has_valid_shape,
-        has_valid_column_number,
-        is_rank_correct
-    ))
+    result = Result(result_df)
+    assert (
+            result == ScoreMatcher()
+            and result == RankDTypeMatcher()
+            and result == RankMonotonicMatcher(n_rows=n_rows, results_per_row=top_k)
+            and result == ShapeMatcher(columns=columns, new_columns=new_columns, n_rows=n_rows, results_per_row=top_k)
+            and result == ColumnsMatcher(columns=columns[1:], new_columns=new_columns)
+            and result == NoErrorMessageMatcher()
```

Comment:
Is this the "pythonic" way of using matchers? Looks a bit strange for Java devs ;)